common/src/main/java/com/wynntils/features/combat/QuickCastFeature.java

```diff
@@ -49,55 +52,44 @@ public class QuickCastFeature extends Feature {
     @RegisterKeyBind
     private final KeyBind castFourthSpell = new KeyBind("Cast 4th Spell", GLFW.GLFW_KEY_V, true, this::castFourthSpell);
 
+    @RegisterConfig
+    private final Config<Integer> rightClickCooldown = new Config<>(3);
+
+    @RegisterConfig
+    private final Config<Boolean> safeCasting = new Config<>(true);
+
+    @RegisterConfig
+    private final Config<Integer> spellCooldown = new Config<>(2);
+
     private static final Pattern INCORRECT_CLASS_PATTERN = compileCCRegex("§✖§ Class Req: (.+)");
     private static final Pattern LVL_MIN_NOT_REACHED_PATTERN = compileCCRegex("§✖§ (.+) Min: ([0-9]+)");
 
-    private SpellDirection[] spellInProgress = SpellDirection.NO_SPELL;
-
-    private static final Queue<Runnable> SPELL_PACKET_QUEUE = new LinkedList<>();
+    private static final Queue<SpellDirection> SPELL_PACKET_QUEUE = new LinkedList<>();
 
+    private final Queue<SpellDirection> currentSpell = new LinkedList<>();
+    private final Queue<Spell> spells = new LinkedList<>();
+    private Spell spell = null;
     private int packetCountdown = 0;
     private int spellCountdown = 0;
     private int lastSelectedSlot = 0;
 
-    @SubscribeEvent
-    public void onSpellSequenceUpdate(SpellEvent.Partial e) {
-        updateSpell(e.getSpellDirectionArray());
-    }
-
-    private void updateSpell(SpellDirection[] spell) {
-        if (Arrays.equals(spellInProgress, spell)) return;
-        if (spell.length == 3) {
-            spellInProgress = SpellDirection.NO_SPELL;
-            spellCountdown = 0;
-        } else {
-            spellInProgress = spell;
-            spellCountdown = 40;
-        }
-    }
-
     private void castFirstSpell() {
-        tryCastSpell(SpellUnit.PRIMARY, SpellUnit.SECONDARY, SpellUnit.PRIMARY);
+        tryCastSpell(SpellUnit.SECONDARY, SpellUnit.PRIMARY);
     }
 
     private void castSecondSpell() {
-        tryCastSpell(SpellUnit.PRIMARY, SpellUnit.PRIMARY, SpellUnit.PRIMARY);
+        tryCastSpell(SpellUnit.PRIMARY, SpellUnit.PRIMARY);
     }
 
     private void castThirdSpell() {
-        tryCastSpell(SpellUnit.PRIMARY, SpellUnit.SECONDARY, SpellUnit.SECONDARY);
+        tryCastSpell(SpellUnit.SECONDARY, SpellUnit.SECONDARY);
     }
 
     private void castFourthSpell() {
-        tryCastSpell(SpellUnit.PRIMARY, SpellUnit.PRIMARY, SpellUnit.SECONDARY);
+        tryCastSpell(SpellUnit.PRIMARY, SpellUnit.SECONDARY);
     }
 
-    private void tryCastSpell(SpellUnit a, SpellUnit b, SpellUnit c) {
-        if (!SPELL_PACKET_QUEUE.isEmpty()) {
-            sendCancelReason(Component.translatable("feature.wynntils.quickCast.anotherInProgress"));
-            return;
-        }
-
+    private void tryCastSpell(SpellUnit a, SpellUnit b) {
```

Comment:
I don't appreciate that you removed the first part of the spells. This is really unclear to anyone reading this not knowing it's history. I think this method should take in the full spell combo, even if it is unused. I think @magicus agrees?